TLM/TLM/UI/Helpers/DropDownOption.cs

```diff
@@ -0,0 +1,98 @@
+namespace TrafficManager.UI.Helpers {
+    using ICities;
+    using ColossalFramework.UI;
+    using TrafficManager.State;
+    using CSUtil.Commons;
+    using System;
+    using System.Linq;
+    using TrafficManager.API.Traffic.Enums;
+
+    public class DropDownOption<TEnum> : SerializableUIOptionBase<TEnum, UIDropDown, DropDownOption<TEnum>>
+        where TEnum : struct, Enum, IConvertible {
+
+        public DropDownOption(string fieldName, Options.PersistTo scope = Options.PersistTo.Savegame)
+        : base(fieldName, scope) { }
+
+        /* Data */
+        private static TEnum[] values_ = Enum.GetValues(typeof(TEnum)) as TEnum[];
+
+        private static string[] names_ = Enum.GetNames(typeof(TEnum));
+
+        private static string[] keys_ =
+            names_.
+            Where(item => item != "MaxValue").
+            Select(LocaleKeyAttribute.GetKey<TEnum>).
+            ToArray();
+
+        private string[] GetTranslatedItems() =>
+            keys_.Select(Translate).ToArray();
+
+        private static int IndexOf(TEnum value) =>
```

Comment:
Should this be instance-based?

Reply:
I think if it works without instance then it should not be instance based.